### PR TITLE
(PC-7937) fix sign in modal for small devices and header notch

### PR DESCRIPTION
**Pull request by emmajmn**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-7937

## Checklist

I have:

- [x] Made sure the title of my PR follows the convention `[PC-XXX] <summary>`.
- [x] Made sure my feature is working on the relevant real / virtual devices.
- [ ] Written **unit tests** for my feature.
- [ ] Added new reusable components to the AppComponents page and communicate to the team on slack
- [x] Added a **screenshot** for UI tickets.
- [ ] Attached a **ticket number** for any added TODO/FIXME \
       (for tech tasks, give the best context about the TODO resolution: what/who/when).

## Screenshots

| \*iOS - [Phone name] | \*Android - [Phone name] | Tablet - [Phone name] | Little - [Phone name] |
| -------------------- | :----------------------: | --------------------: | --------------------: |
| [video](https://user-images.githubusercontent.com/50346870/114726554-ca728200-9d3d-11eb-822a-0760a1ef092d.mov) |                          |                       |                       |

## Deploy hard

If native code (ios/android) was modified, **after** the PR is merged, on the master branch, upgrade the **app version** (+1 patch):

- if you want an hard deployment of the testing environment, use `yarn trigger:testing:deploy`

### CODE REVIEW

**Review by kopax-polyconseil — APPROVED** [review]

File: src/ui/components/BottomContentPage.tsx [hunk 1]
```diff
@@ -37,37 +36,45 @@ export const BottomContentPage: FC<Props> = (props) => {
...
 }
 
+const BottomContentPageContainer = styled(SafeAreaView)({
+  height: '100%',
+})
+
 const Container = styled(View).attrs({
   onPress: Keyboard.dismiss,
 })({
-  zIndex: 41,
```
Comment (blixit) [inline src/ui/components/BottomContentPage.tsx:57]: t'as bien mesuré l'impact de la suppression du zIndex ?

File: src/ui/components/BottomContentPage.tsx [hunk 2]
```diff
@@ -37,37 +36,45 @@ export const BottomContentPage: FC<Props> = (props) => {
...
-    <React.Fragment>
+    <BottomContentPageContainer>
       <Background />
       <Container>
-        <AvoidingKeyboardContainer keyboardHeight={keyboardHeight}>
-          <StyledBottomCardContainer>
+        <KeyboardAvoidingView behavior={Platform.OS === 'ios' ? 'padding' : 'height'}>
+          <StyledBottomCardContainer
+            // eslint-disable-next-line react-native/no-inline-styles
+            contentContainerStyle={{
```
Comment (blixit) [inline src/ui/components/BottomContentPage.tsx:45]: pourquoi du style inline ?

Comment (emmajmn) [reply src/ui/components/BottomContentPage.tsx:45]: le style dépends d'une variable du composant du coup je trouvais plus pertinent de l'avoir comme ça ? tu as une reco sinon ?

File: src/ui/components/BottomContentPage.tsx (hunk 1 above)
Comment (emmajmn) [reply src/ui/components/BottomContentPage.tsx:57]: yes ! J'ai en plus vérifié tous les endroits ou le composant BottomContentPage était appelé sur les différents OS et tout fonctionne parfaitement :)

File: src/ui/components/BottomContentPage.tsx (hunk 2 above)
Comment (blixit) [reply src/ui/components/BottomContentPage.tsx:45]: voici un exemple de pattern que tu pourrais utiliser:

```ts
// NonBeneficiaryHeaderComponent.tsx

const BodyContainer = styled.View<{ padding: number }>(({ padding }) => ({
  padding,
}))

```

**Review by blixit — APPROVED** [review]

**Pull request merged**